meetings/2022-09-28.md

```diff
@@ -0,0 +1,187 @@
+* Elf: But isn't the deny a default?
+* Matthieu: The default is that nothing is allowed.
+* ...: Not that everything is denied.
+* ...: Deny acts as a thing that trumps everything that is allowed (something denied cannot be otherwise allowed).
+* Elf: Let's look for example at the [client constraint evaluation](https://github.com/solid/authorization-panel/blob/main/proposals/evaluation/uc-2.5.2-client-constraints.md) (simplified): 
+
+```ttl
+PREFIX acp: <http://www.w3.org/ns/solid/acp#>
+PREFIX acl: <http://www.w3.org/ns/auth/acl#>
+PREFIX ex: <https://example.com/>
+
+ex:ReaderAgentPolicy
+  acp:anyOf ex:BobMatcher ;
+  acp:allow acl:Read .
+
+ex:BobMatcher
+  acp:agent ex:Bob ;
+  acp:client ex:Projectron .
+
+```
+
+* Elf: and discuss the difference between those two matchers:
+
+```ttl
+ex:BobMatcher
+  acp:agent ex:Bob ;
+  acp:client acp:PublicClient .
+```
+
+```ttl
+ex:BobMatcher
+  acp:agent ex:Bob .
+```
+
+* Matthieu: The matcher where Bob and the ~~Public Client~~ Authenticated Client (Note to reader: I thought we were discussing Authenticated agent) are required to satisfy authorization will require Bob using a client.
+* ...: The other matcher will be satisfied if Bob does not use a client as well.
+* Elf: Could you write the second case context
+
+```ttl
+[]
+  acp:target ex:resourceX ;
+  acp:agent ex:Bob .
+```
+
+* Matthieu: The above context being client-less will resolve successfully if acp:client acp:PublicClient is not required.
+* ...: Really it is just a case of unmitigated resource access. If we say unmitigated resource access exists, then the AuthZ domain needs to be able to express this.
+* Matthieu: Sorry I meant "for acp:AuthenticatedClient", I thought we were discussing unmitigated resource access.
+* Elf: In ACP if you ommit acp:agent in the matcher, any agent (incl. no agent) will match? Assuming other properties in the matcher.
+* Matthieu: An empty matcher is never satisfied.
+* ...: This issue [312](https://github.com/solid/authorization-panel/issues/312) is really just about acknowledging the existence of unmitigated client access.
+* ...: I agree that the "Public" named individuals are a bit misleading because they are not matching on a specific attribute, we could consider creating an always matcher instead...
+* Elf: Ok, I see how AuthenticatedClient makes sense.
+* ...: So as soon as you have a non-empty matcher the attributes targetting a "Public" named individual can be ignored?
+* Matthieu: Yes. And empty matchers are never satisfied.
+* ...: For example, the following Policy will never be satisfied because the empty matcher of the allOf condition will never be satisfied.
+
+```ttl
+[]
+  acp:allow acl:Read ;
+  acp:allOf [ acp:agent ex:Bob ], [] ;
+```
+
+* Matthieu: On the other hand, in the following Policy, the second matcher adds no value. It will be satisfied if Bob is the agent.
+
+```ttl
+[]
+  acp:allow acl:Read ;
+  acp:allOf [ acp:agent ex:Bob ], [ acp:client acp:PublicClient ] ;
+```
```

Comment:
As above, every line from 165 to 169 needs to be indented with 2 (additional) space chars, so that block gets rendered under/within Matthieu's `agent.` bullet. I cannot submit this as a suggestion, because GitHub can't handle the 3-backtick code fences within a suggestion.